azurelinuxagent/common/event.py

```diff
@@ -339,12 +342,27 @@ def add_metric(self, category, counter, instance, value, log_event=False):
         event.parameters.append(TelemetryEventParam('Instance', instance))
         event.parameters.append(TelemetryEventParam('Value', value))
 
+        self._add_default_parameters_to_event(event, container_id=os.environ.get(
+            CONTAINER_ID_ENV_VARIABLE, "UNINITIALIZED"), event_time_opcode_name=event_time_opcode_name,
+                                              event_pid=event_pid, event_tid=event_tid, task_name=task_name,
+                                              keyword_name=keyword_name)
         data = get_properties(event)
         try:
             self.save_event(json.dumps(data))
         except EventError as e:
             logger.error("{0}", e)
 
+    @staticmethod
+    def _add_default_parameters_to_event(event, container_id="UNINITIALIZED", event_time_opcode_name=None, event_pid=0,
```

Comment:
Can you extract `TelemetryEventParam("GAVersion", CURRENT_AGENT)` from `init_sysinfo` to here since it should be retrieved and populated when we are constructing the event, not only once on agent startup?